spec/osut_tests_spec.rb

```diff
@@ -1298,4 +1300,503 @@ module M
+    gap    = 0.200
+    frame  = 0.050
+    frames = 2 * frame
+
+    fd = OpenStudio::Model::WindowPropertyFrameAndDivider.new(model)
+    expect(fd.setFrameWidth(frame)).to be(true)
+    expect(fd.setFrameConductance(2.500)).to be(true)
+
+    w1              = {}
+    w1[:id        ] = "OA1 W4 w1"
+    w1[:frame     ] = fd
+    w1[:width     ] = 0.8
+    w1[:head      ] = HEAD
+    w1[:sill      ] = 0.005 + frame # to avoid generating a warning
+    w1[:centreline] = -xmid + gap + frame + w1[:width]/2
+
+    w2              = {}
+    w2[:id        ] = "OA1 W4 w2"
+    w2[:frame     ] = fd
+    w2[:width     ] = w1[:width     ]/2
+    w2[:head      ] = w1[:head      ]
+    w2[:sill      ] = w1[:sill      ]
+    w2[:centreline] = w1[:centreline] + w1[:width]/2 + frames + w2[:width]/2
+
+    t1              = {}
+    t1[:id        ] = "OA1 W4 t1"
+    t1[:frame     ] = fd
+    t1[:width     ] = w1[:width     ]
+    t1[:height    ] = w2[:width     ]
+    t1[:sill      ] = w1[:head      ] + frames
+    t1[:centreline] = w1[:centreline]
+
+    t2              = {}
+    t2[:id        ] = "OA1 W4 t2"
+    t2[:frame     ] = fd
+    t2[:width     ] = w2[:width     ]
+    t2[:height    ] = t1[:height    ]
+    t2[:sill      ] = t1[:sill      ]
+    t2[:centreline] = w2[:centreline]
+
+    sbz = [w1, w2, t1, t2]
+    expect(mod1.addSubs(model, wall4, sbz)).to be(true)
+    expect(mod1.status.zero?).to be(true)
+
+
+    # Add another 5x (frame&divider-enabled) fixed windows, from either
+    # left- or right-corner of base surfaces. Fetch "Openarea Wall 6".
+    wall6 = model.getSurfaceByName("Openarea 1 Wall 6")
+    expect(wall6.empty?).to be(false)
+    wall6 = wall6.get
+
+    # Fetch "Openarea Wall 7".
+    wall7 = model.getSurfaceByName("Openarea 1 Wall 7")
+    expect(wall7.empty?).to be(false)
+    wall7 = wall7.get
+
+    # Fetch 'head'/'sill' heights of nearby "Sub Surface 6".
+    sub6 = model.getSubSurfaceByName("Sub Surface 6")
+    expect(sub6.empty?).to be(false)
+    sub6 = sub6.get
+    sub6_min = sub6.vertices.map(&:z).min
+    sub6_max = sub6.vertices.map(&:z).max
+
+    # 1x Array of 3x windows, 8" from the left corner of wall6.
+    a6              = {}
+    a6[:id        ] = "OA1 W6 a6"
+    a6[:count     ] = 3
+    a6[:frame     ] = fd
+    a6[:head      ] = sub6_max
+    a6[:sill      ] = sub6_min
+    a6[:width     ] = a6[:head ] - a6[:sill]
+    a6[:offset    ] = a6[:width] + gap
+    a6[:l_buffer  ] = gap
+
+    expect(mod1.addSubs(model, wall6, [a6])).to be(true)
+    expect(mod1.status.zero?).to be(true)
+
+    # 1x Array of 2x square windows, 8" from the right corner of wall7.
+    a7              = {}
+    a7[:id        ] = "OA1 W6 a7"
+    a7[:count     ] = 2
+    a7[:frame     ] = fd
+    a7[:head      ] = sub6_max
+    a7[:sill      ] = sub6_min
+    a7[:width     ] = a7[:head ] - a7[:sill]
+    a7[:offset    ] = a7[:width] + gap
+    a7[:r_buffer  ] = gap
```

Comment:
Tests left/right buffer-alignment (instead of _centred_ or _off-centred_) of 5x ~square-shaped windows, in the 2x (fairly narrow) walls in the centre of the Figure:

1. 1x array of 2x windows added to the left wall ("right-aligned")
2. 1x array of 3x windows added to the right wall ("left-aligned)

<img width="1352" alt="leftright" src="https://github.com/rd2/osut/assets/98104985/3a64902c-fc9f-47ba-98f2-531d736a693b">

A "gap" (200mm or 8") separates the glazing areas between each other, and between the left (or right) wall corner and the nearest window glazing edge. EnergyPlus simulations run fine (no related warnings/errors).